hw04-Log/src/main/java/ru.otus/IoC.java

```diff
@@ -0,0 +1,54 @@
+package ru.otus;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IoC {
+
+    static TestInterface createMyClass() {
+        InvocationHandler handler = new DemoInvocationHandler(new TestLogging());
+        return (TestInterface) Proxy.newProxyInstance(IoC.class.getClassLoader(),
+                new Class<?>[]{TestInterface.class}, handler);
+    }
+
+    static class DemoInvocationHandler implements InvocationHandler {
+        private final TestInterface myClass;
+        private final Set<String> listMethods = new HashSet<>();
```

Comment:
Правильно, что применили HashSet.
Только в имени поля `listMethods` намек на то, что у вас список (List), а на самом деле у вас множество (Set). Лучше переименовать.

```suggestion
        private final Set<String> methodsForLog = new HashSet<>();
```